cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/web/SearchController.java

```diff
@@ -0,0 +1,63 @@
+package edu.asu.diging.cord19.explorer.web;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import edu.asu.diging.cord19.explorer.core.model.impl.PublicationImpl;
+import edu.asu.diging.cord19.explorer.core.mongo.AffiliationSearchProvider;
+import edu.asu.diging.cord19.explorer.core.mongo.PublicationSearchProvider;
+import edu.asu.diging.cord19.explorer.core.mongo.impl.AffiliationPaperAggregationOutput;
+import edu.asu.diging.cord19.explorer.web.model.SearchType;
+
+@Controller
+public class SearchController {
+
+    
+    @Autowired
+    private AffiliationSearchProvider affSearchProvider;
+    
+    @Autowired
+    private PublicationSearchProvider pubSearchProvider;
+    
+    
+    
+    
+    @RequestMapping(value = "/search")
+    public String search(@RequestParam("searchType") String searchType,@RequestParam("search") String query, Model model,
```

Comment:
I think I would also add the path `/{pathSearchType}/search` (or something like that) and then have a not required pathVariable that holds the searchType (and make `searchType` not required). this way on the specific result pages, you can just make calls to `/affiliations/search` or `/publications/search` to move to the next page to make it easier to use/remember/etc.